sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/sources/ForeachWriterTable.scala

```diff
@@ -142,21 +142,27 @@ class ForeachDataWriter[T](
       writer.process(rowConverter(record))
     } catch {
       case t: Throwable =>
-        writer.close(t)
-        closed = true
+        closeWriter(t)
         throw t
     }
   }
 
   override def commit(): WriterCommitMessage = {
-    writer.close(null)
-    closed = true
+    closeWriter(null)
     ForeachWriterCommitMessage
   }
 
   override def abort(): Unit = {
-    if (!closed) {
-      writer.close(new RuntimeException("Foreach writer has been aborted"))
+    closeWriter(new RuntimeException("Foreach writer has been aborted"))
+  }
+
+  private def closeWriter(errorOrNull: Throwable): Unit = {
+    if (!closeCalled) {
+      try {
+        writer.close(errorOrNull)
+      } finally {
```

Comment:
Is the exception coming from `writer.close` printed out somewhere? Maybe it's valuable info for the users.

Reply:
i have to agree with @gaborgsomogyi , it will be valuable to log the error here

Reply:
Since writer.close is user-facing API and users know exception is passed to the parameter, they have a chance to add log in writer implementation.

Btw, I feel logging is depending on preference, like which level, what data to leave, etc. Committers might think differently and they may not go through the discussion on here.

Could we merge this in and push, and let @eyalzit add a new commit to log message?

Reply:
@HeartSaVioR sure, I meant the same but not said explicitly.